tests/nightly/test_large_array.py

```diff
@@ -35,1658 +35,1627 @@
+                         lambda x: 1.0 / (1.0 + np.exp(-x)),
+                         lambda x, y: x - y,
+                         shape)
+        check_regression(mx.symbol.LinearRegressionOutput,
+                         lambda x: x,
+                         lambda x, y: x - y,
+                         shape)
+
+    def check_l2_normalization():
+        x = nd.ones((2, LARGE_X*2))
+        x[0] = 3
+        x[1] = 4
+        # Channel Mode
+        z = x.reshape(1, 2, LARGE_X*2)
+        y = nd.L2Normalization(z, mode='channel')
+        assert y[0][0][0] == 0.6
+        assert y[0][0][-1] == 0.6
+        assert y[0][1][0] == 0.8
+        assert y[0][1][-1] == 0.8
+        # Instance Mode
+        z = x.T
+        y = nd.L2Normalization(z, mode='instance')
+        assert y[0][0] == 0.6
+        assert y[0][1] == 0.8
+        assert y[-1][0] == 0.6
+        assert y[-1][1] == 0.8
+        # Spatial Mode
+        z = z.reshape(1, 200000000, 2)
+        y = nd.L2Normalization(z, mode='spatial')
+        assert y[0][0][0] == 0.6
+        assert y[0][0][1] == 0.8
+        assert y[0][-1][0] == 0.6
+        assert y[0][-1][1] == 0.8
+
+    def check_instance_norm():
+        dtype = np.float32
+        forward_check_eps = 1E-3
+        axis = -1
+        eps = 1E-5
+        in_shape = (LARGE_X, 1, SMALL_Y)
+        ctx = mx.cpu()
+
+        # Implementation of instance normalization using numpy
+        def npy_instance_norm(data, gamma, beta, axis, eps=1E-5):
+            if axis < 0:
+                axis += data.ndim
+            broadcast_shape = [1 for _ in range(data.ndim)]
+            broadcast_shape[axis] = data.shape[axis]
+            mean = data.mean(axis=axis, keepdims=True).astype(dtype)
+            var = data.var(axis=axis, keepdims=True).astype(dtype)
+            std = np.sqrt(var + dtype(eps)).astype(dtype)
+            out = gamma * (data - mean) / std + \
+                  beta
+            return out
+        data = np.random.normal(0, 1, in_shape).astype(dtype)
+        gamma = np.random.normal(0, 1, (1,)).astype(dtype)
+        beta = np.random.normal(0, 1, (1,)).astype(dtype)
+        data_s = mx.symbol.Variable('data')
+        gamma_s = mx.symbol.Variable('gamma')
+        beta_s = mx.symbol.Variable('beta')
+        out_s = mx.symbol.InstanceNorm(data=data_s, gamma=gamma_s, beta=beta_s,
+                                       eps=eps)
+        exe = out_s.simple_bind(ctx, data=in_shape)
+        exe.arg_dict['data'][:] = data
+        exe.arg_dict['gamma'][:] = gamma
+        exe.arg_dict['beta'][:] = beta
+        out_nd = exe.forward()[0]
+        # Calls implementation of instance norm in numpy and compares the output
+        out = npy_instance_norm(data, gamma, beta, axis, eps)
+        assert_almost_equal(out, out_nd.asnumpy(), forward_check_eps,
+                            forward_check_eps)
+
+    check_gluon_embedding()
+    check_fully_connected()
+    check_dense(ctx=mx.cpu(0))
```

Comment:
Why hard code ctx to cpu? Is it too big for GPU?

Reply:
my bad. Copy paste error. https://github.com/apache/incubator-mxnet/pull/17305/files#diff-9ee911616af04047075035c95cf542fbR60 already has default parameter as cpu ctx. 

We are not testing for GPU since theoretically its not possible for a 32-bit(4 Byte) variable to hold such a large value.

```
(2^32*4)/(2^30) = 16 GB
``` 
Unless its v100 w/ 32 GB Global mem.